File: python_backend/game_logic/threatArea/get_king_loc.py
from game_logic.test_objects.sample_board_dicts import sample_board_dicts


def get_king_loc(board, color):
    """return the location of the king of specified color"""

    for coord, id in board.items():
        if id[0] == color and id[1] == 'K':
            return coord
    else:
        print('error: king not found\n')
        return -1


if __name__ == "__main__":
    print('test 1:')
    board = sample_board_dicts['check_example3']
    king_loc = get_king_loc(board, 'W')
    print(king_loc == (4, 4))
    print('\n')

File: app/src/game_logic/threatArea/getKingLoc.js
import { rfToXy } from "../coordType/crdCnvrt";
import { getPieceType } from "../pieceType/getPieceType";
import { getColor } from "../color/getColor";

export function getKingLoc(board, color) {
    /*get the location of the king of given color NOTE: in the modified chess can return list instead of single item**/
    var locs = Object.keys(board).filter(
        (s) => getPieceType(board[s]) === "K" && getColor(board[s]) === color
    );
    return rfToXy(locs[0]);
}

// module.exports = getKingLoc;

/** for node.js
if (require.main === module) {
    
    import {sampleBoardDicts} from "../testObjects/sampleBoardDicts";
    import {printBoard} from "../printers/printBoard";

    var board, kingLoc;
    
    board = sampleBoardDicts['check_example3']
    kingLoc = getKingLoc(board, 'W')
    printBoard(board, `\ncheck_example3, hover-off`, [kingLoc])
    kingLoc = getKingLoc(board, 'B')
    printBoard(board, `\ncheck_example3, black`, [kingLoc])

    board = sampleBoardDicts['super_checkmate_impossible_example']
    kingLoc = getKingLoc(board, 'W')
    printBoard(board, `\nsuper_checkmate_impossible_example, hover-off`, [kingLoc])
    kingLoc = getKingLoc(board, 'B')
    printBoard(board, `\nsuper_checkmate_impossible_example, black`, [kingLoc])
}
*/
